### PR TITLE
Refactor long running SQL schema version tests to reduce runtime

### PR DESCRIPTION
**Pull request by apurvabhaleMS**
## Description
With this PR we are using the Parallel ForEach for the long-running SQL schema version tests. 

SqlServerSqlCompatibilityTests.GivenADatabaseWithAnEarlierSupportedSchema_WhenUpserting_OperationSucceeds
	- Looping through the versions from min to max. Creating a database for each version and sending a post request to fhir server.
	
SqlServerSchemaUpgradeTests.GivenASchemaVersion_WhenApplyingDiffTwice_ShouldSucceed
	- Gets called for each version, Creates a new DB with calledVersion-1 version then tries to apply the called version twice

## Related issues
Addresses [90156](https://microsofthealth.visualstudio.com/Health/_boards/board/t/Olympus/Stories/?workitem=90156)

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs [hunk 1]
```diff
@@ -75,42 +75,34 @@ public async Task GivenTwoSchemaInitializationMethods_WhenCreatingTwoDatabases_B
...
-            }
-        }
+                    SqlServerFhirStorageTestHelper testHelper = null;
+                    SchemaUpgradeRunner upgradeRunner;
 
-        public static IEnumerable<object[]> GetSchemaVersions()
-        {
-            foreach (object item in Enum.GetValues(typeof(SchemaVersion)))
-            {
-                // The schema upgrade scripts starting from v7 were made idempotent.
```
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs:107]: Could you add this comment back to the if statement on line 83? I think it is important to remember why version 7 is important.

**Comment by LTA-Thinking** [comment]
I'm a little concerned that the time to run these tests has dropped from 10s of minutes to < 3ms. I was expecting a performance bump, but this makes it look like the tests aren't running properly.

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs (hunk 1 above)
Comment (apurvabhaleMS) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs:107]: Yes, I added the comment

**Comment by apurvabhaleMS** [comment]
> I'm a little concerned that the time to run these tests has dropped from 10s of minutes to < 3ms. I was expecting a performance bump, but this makes it look like the tests aren't running properly.

You are right, the versions were not retrieved correctly. Latest commit should fix this

**Pull request merged**